packages/@lwc/compiler/src/transformers/template.ts

```diff
@@ -71,9 +73,16 @@ export default function templateTransform(
         throw normalizeToCompilerError(TransformerErrors.HTML_TRANSFORMER_ERROR, e, { filename });
     }
 
-    const fatalError = result.warnings.find((warning) => warning.level === DiagnosticLevel.Error);
-    if (fatalError) {
-        throw CompilerError.from(fatalError, { filename });
+    const errors = result.warnings.filter((warning) => warning.level === DiagnosticLevel.Error);
+
+    if (experimentalErrorRecoveryMode && errors.length > 0) {
+        throw new CompilerAggregateError(
+            errors.map((err) => CompilerError.from(err, { filename }))
+        );
+    }
+
+    if (errors[0]) {
```

Comment:
```suggestion
    // If we're not in error recovery mode, then we will only ever have zero or one errors
    if (errors[0]) {
```

Reply:
That's not really true.

The way template-compiler currently works (even without explicit errorRecoveryMode) is that it runs parsing of `root` and `children` in a protected boundary so even if one errors out we store that error and skip it's sub-tree but continue parsing rest of the AST.

Even though we have all the errors available we [Array.find](https://github.com/salesforce/lwc/pull/5582/files#diff-7901555acef29969adaa6583185b3e9bce475cdc6f23e799a54e0018cb18abaaL74) the first error and throw, not sure why though :)